File: backend/catalogo/src/shared/AggregateRoot.js
import { Entity } from "./core/Entity.js";
import { BusinessRuleValidationException } from "./BusinessRuleValidationException.js";

export class AggregateRoot extends Entity {
  constructor(id) {
    super(id);
  }

  static checkRule(rule) {
    if (!rule.isValid()) {
      throw new BusinessRuleValidationException(rule);
    }
  }
}

File: backend/pagos/payments/domain/shared/core/aggregate_root.py
from .entity import Entity
from .business_rule import BusinessRule
from .business_rule_validation_exception import BusinessRuleValidationException

class AggregateRoot(Entity):
    def __init__(self, entity_id):
        super().__init__(entity_id)

    @staticmethod
    def check_rule(rule: BusinessRule):
        if not rule.is_valid():
            raise BusinessRuleValidationException(rule)